### PR TITLE
[I18N] delivery_grid: improved translation

### PR DESCRIPTION
**Pull request by stcc-odoo**
Steps to reproduce:
- Install eCommerce, delivery
- Shpping methods > Create > Name = TEST, Product = Acoustic Bloc Screen, Provider = Based on rules, Add a line > Condition = "Weight <= 0", Delivery Cost = "50 + 0 * Weight" > Publish shipping method
- Go to the website > Add acoustic Bloc Screen to cart > Proceed to checkout

In the delivery section, the TEST shipping method will display the warning message `No price rule matching this order; delivery cost cannot be computed.` This message can be confusing to the customer.

opw-3110715

### CODE REVIEW

**Comment by stcc-odoo** [comment]
BEFORE
image
AFTER
image

**Review by DjamelTouati — CHANGES_REQUESTED** [review]
I think it's better to make this kind of change in master, because in stable we will lose all the translations. I agree that the message can be confusing, but it is not wrong, so I suggest you to target master instead

File: addons/delivery/i18n/delivery.pot [hunk 1]
```diff
@@ -655,7 +655,7 @@ msgstr ""
 #. module: delivery
 #: code:addons/delivery/models/delivery_grid.py:0
 #, python-format
-msgid "No price rule matching this order; delivery cost cannot be computed."
+msgid "Not available for current order"
```
Comment (nle-odoo) [inline addons/delivery/i18n/delivery.pot:658]: should be below "Normal Delivery Charges" I think (but in master I think it's not necessary)

**Comment by stcc-odoo** [comment]
Superseded by https://github.com/odoo/odoo/pull/112002

**Pull request closed without merging**